src/OnlineSales/Controllers/ImagesController.cs

```diff
@@ -0,0 +1,118 @@
+﻿// <copyright file="ImagesController.cs" company="WavePoint Co. Ltd.">
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking.Internal;
+using Microsoft.Net.Http.Headers;
+using OnlineSales.Data;
+using OnlineSales.DTOs;
+using OnlineSales.Entities;
+using Quartz.Util;
+
+namespace OnlineSales.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ImagesController : ControllerBase
+    {
+        private readonly ApiDbContext apiDbContext;
+
+        public ImagesController(ApiDbContext apiDbContext)
+        {
+            this.apiDbContext = apiDbContext;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> Post([FromForm] ImageDtos imageDtos)
+        {
+            var provider = new FileExtensionContentTypeProvider();
+
+            string incomingFileName = imageDtos.Image!.FileName;
```

Comment:
@WASandaruwan you made FileSizeValidateAttribute and FileExtensionValidateAttribute attributes, but as I understand they would never take affect you never check if ModelState.IsValid or not. Please correct me if I am wrong.